server/api/student/student.model.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+var mongoose = require('mongoose'),
+    Schema = mongoose.Schema;
+
+var schoolYearSchema = new Schema({
+	allAbsences: Number,
+	tardy: Number,
+	present: Number,
+	enrolled: Number
+});
+
+var StudentSchema = new Schema({
+  _id: Number, // Student ID
+  lastName: String,
+  firstName: String,
+  schoolYear: [ Number: {schoolYearSchema}]
```

Comment:
Does this work? I haven't seen this declaration pattern before.

I've seen subdocs declared like this: http://mongoosejs.com/docs/subdocs.html

Where you would include the year in the SchoolYearSchema:

```
var SchoolYearSchema = new Schema({
  year: String,
  allAbsences: Number,
  tardy: Number,
  present: Number,
  enrolled: Number
});

var StudentSchema = new Schema({
  _id: Number, // Student ID
  lastName: String,
  firstName: String,
  schoolYears: [SchoolYearSchema]
});
```

I'm still trying to figure out if using subdocuments here is the best solution since we need to do aggregation reporting. Let's go with this for now. We can revisit this soon.